flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/filesystem/stream/compact/CompactCoordinatorTest.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.filesystem.stream.compact;
+
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.checkpoint.OperatorSubtaskState;
+import org.apache.flink.streaming.util.OneInputStreamOperatorTestHarness;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.CompactionUnit;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.CoordinatorInput;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.CoordinatorOutput;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.EndCompaction;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.EndInputFile;
+import org.apache.flink.table.filesystem.stream.compact.CompactMessages.InputFile;
+import org.apache.flink.util.function.ThrowingConsumer;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * Test for {@link CompactCoordinator}.
+ */
+public class CompactCoordinatorTest extends AbstractCompactTestBase {
+
+	@Test
+	public void testCoordinatorCrossCheckpoints() throws Exception {
+		AtomicReference<OperatorSubtaskState> state = new AtomicReference<>();
+		runCoordinator(harness -> {
+			harness.setup();
+			harness.open();
+
+			harness.processElement(new InputFile("p0", newFile("f0", 3)), 0);
+			harness.processElement(new InputFile("p0", newFile("f1", 2)), 0);
+
+			harness.processElement(new InputFile("p1", newFile("f2", 2)), 0);
+
+			harness.processElement(new InputFile("p0", newFile("f3", 5)), 0);
+			harness.processElement(new InputFile("p0", newFile("f4", 1)), 0);
+
+			harness.processElement(new InputFile("p1", newFile("f5", 5)), 0);
+			harness.processElement(new InputFile("p1", newFile("f6", 4)), 0);
+
+			state.set(harness.snapshot(1, 0));
+		});
+
+		runCoordinator(harness -> {
+			harness.setup();
+			harness.initializeState(state.get());
+			harness.open();
+
+			harness.processElement(new InputFile("p0", newFile("f7", 3)), 0);
+			harness.processElement(new InputFile("p0", newFile("f8", 2)), 0);
+
+			state.set(harness.snapshot(2, 0));
+		});
+
+		runCoordinator(harness -> {
+			harness.setup();
+			harness.initializeState(state.get());
+			harness.open();
+
+			harness.processElement(new EndInputFile(2, 0, 1), 0);
+
+			List<CoordinatorOutput> outputs = harness.extractOutputValues();
+
+			Assert.assertEquals(7, outputs.size());
+
+			assertUnit(outputs.get(0), 0, "p0", Arrays.asList("f0", "f1", "f4"));
```

Comment:
I think `CompactCoordinator` doesn't guarantee to generate `CompactionUnit` in any specific order of partitions, right?

Reply:
You mean the order of partitions? There is no relationship between partitions, so there is no need to guarantee this.

Reply:
Yeah... but then how could we assert the first output is for `p0`?

Reply:
You mean we should sort it before assert?

Reply:
OK, I'll do it